lib/mocha/object_methods.rb

```diff
@@ -10,6 +10,18 @@ module Mocha
   #
   # Both {#expects} and {#stubs} return an {Expectation} which can be further modified by methods on {Expectation}.
   module ObjectMethods
+    if RUBY_PLATFORM == 'java'
+      # @private
+      module JRubyAliasMethodWarningFilter
+        def warn(message, *args)
+          super unless /ObjectMethods#method accesses caller method's state and should not be aliased/.match?(message)
+        end
+      end
+
+      # using public_send as calling extend directly causes YARD to warn about Undocumentable mixin
+      Warning.public_send(:extend, JRubyAliasMethodWarningFilter) # rubocop:disable Lint/SendWithMixinArgument, Style/SendWithLiteralMethodName
```

Comment:
This looks a bit ugly. Could we avoid the `public_send` workaround (and associated comment) and the disabling of two cops by using an approach more like [this](https://github.com/freerange/mocha/blob/d07e12e192ea5a2b9694a35b429308d12dba0a1b/test/unit/object_inspect_test.rb#L68-L78)?

As an aside, I wonder if it might separately be worth trying to extract this pattern somewhere. However, if so, I'd prefer to do that in a separate PR.

Reply:
Addressed in https://github.com/freerange/mocha/pull/741/commits/b578fdcbf134260dba82e6cec4826b310761e8b7 & https://github.com/freerange/mocha/pull/741/commits/3dad2198034d058df236bce90e6dd860f9344bc7.